packages/api/tests/lambdas/test-clean_pg_executions.js

```diff
@@ -0,0 +1,196 @@
+      final_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(2, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(23, 'days').toDate(),
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(10, 'days').toDate(),
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(5, 'days').toDate(),
+      original_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(15, 'days').toDate(),
+      original_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(2, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(23, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(10, 'days').toDate(),
+      original_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(5, 'days').toDate(),
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(15, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+    }),
+    fakeExecutionRecordFactory({
+      updated_at: moment().subtract(2, 'days').toDate(),
+      final_payload: '{"a": "b"}',
+    }),
+  ];
+  const model = new ExecutionPgModel();
+  const inserted = await model.insert(t.context.knex, executions, '*');
+  let expirable = await getExpirablePayloadRecords(t.context.knex, moment().subtract(10, 'days').toDate(), 100);
+  let expected = [
+    inserted[0],
+    inserted[2],
+    inserted[7],
+    inserted[9],
+    inserted[10],
+    inserted[12],
+  ];
+
+  t.deepEqual(expected, expirable);
+  expirable = await getExpirablePayloadRecords(t.context.knex, moment().subtract(0, 'days').toDate(), 100);
+  expected = [
+    inserted[0],
+    inserted[1],
+    inserted[2],
+    inserted[3],
+    inserted[6],
+    inserted[7],
+    inserted[8],
+    inserted[9],
+    inserted[10],
+    inserted[12],
+    inserted[13],
+  ];
+  t.deepEqual(expected, expirable);
+
+  expirable = await getExpirablePayloadRecords(t.context.knex, moment().subtract(3, 'days').toDate(), 100);
+  expected = [
+    inserted[0],
+    inserted[1],
+    inserted[2],
+    inserted[6],
+    inserted[7],
+    inserted[9],
+    inserted[10],
+    inserted[12],
+  ];
+  t.deepEqual(expected, expirable);
+});
+const pgPayloadsEmpty = (entry) => !entry.final_payload && !entry.orginal_payload;
+
+test('cleanupExpiredPGExecutionPayloads() cleans up expired payloads', async (t) => {
+  const env = clone(process.env);
+  process.env = localStackConnectionEnv;
+  process.env.PG_DATABASE = t.context.testDbName;
+  const completeExpirationDays = 3;
+  const nonCompleteExpirationDays = 5;
+
+  const records = [];
+  for (let i = 0; i < 10; i += 1) {
```

Comment:
See previous comment re: creating test executions. Might be able to either share or create a common utility.

Reply:
I believe this should be taken care of with some consolidation of tests